lms/djangoapps/ccx/tests/test_views.py

```diff
@@ -0,0 +1,777 @@
+                parent=chapter,
+                category="sequential",
+                metadata={'graded': True, 'format': 'Homework'})
+            for _ in xrange(4)]
+
+        role = CourseCcxCoachRole(self.course.id)
+        role.add_users(coach)
+        self.ccx = ccx = CcxFactory(course_id=self.course.id, coach=self.coach)
+
+        self.student = student = UserFactory.create()
+        CourseEnrollmentFactory.create(user=student, course_id=self.course.id)
+        CcxMembershipFactory(ccx=ccx, student=student, active=True)
+
+        for i, section in enumerate(sections):
+            for j in xrange(4):
+                item = ItemFactory.create(
+                    parent=section,
+                    category="problem",
+                    data=StringResponseXMLFactory().build_xml(answer='foo'),
+                    metadata={'rerandomize': 'always'}
+                )
+
+                StudentModuleFactory.create(
+                    grade=1 if i < j else 0,
+                    max_grade=1,
+                    student=student,
+                    course_id=self.course.id,
+                    module_state_key=item.location
+                )
+
+        # Apparently the test harness doesn't use LmsFieldStorage, and I'm not
+        # sure if there's a way to poke the test harness to do so.  So, we'll
+        # just inject the override field storage in this brute force manner.
+        OverrideFieldData.provider_classes = None
+        for block in iter_blocks(course):
+            block._field_data = OverrideFieldData.wrap(   # pylint: disable=protected-access
+                coach, block._field_data)   # pylint: disable=protected-access
+            block._field_data_cache = {}  # pylint: disable=protected-access
+            visible_children(block)
+
+        def cleanup_provider_classes():
+            """
+            After everything is done, clean up by un-doing the change to the
+            OverrideFieldData object that is done during the wrap method.
+            """
+            OverrideFieldData.provider_classes = None
+        self.addCleanup(cleanup_provider_classes)
+
+        patch_context = patch('ccx.views.get_course_by_id')
+        get_course = patch_context.start()
+        get_course.return_value = course
+        self.addCleanup(patch_context.stop)
+
+        override_field_for_ccx(ccx, course, 'grading_policy', {
+            'GRADER': [
+                {'drop_count': 0,
+                 'min_count': 2,
+                 'short_label': 'HW',
+                 'type': 'Homework',
+                 'weight': 1}
+            ],
+            'GRADE_CUTOFFS': {'Pass': 0.75},
+        })
+        override_field_for_ccx(
+            ccx, sections[-1], 'visible_to_staff_only', True)
+
+    @patch('ccx.views.render_to_response', intercept_renderer)
+    def test_gradebook(self):
+        url = reverse(
+            'ccx_gradebook',
+            kwargs={'course_id': self.course.id.to_deprecated_string()}
```

Comment:
New code shouldn't use `.to_deprecated_string`. You should just use `unicode(self.course.id)` instead.